### PR TITLE
Modify documentation (olp-cpp-sdk-core/client)

### PR DESCRIPTION
**Pull request by HalynaDumych**
Relates to: OLPEDGE-1441

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h [hunk 1]
```diff
@@ -38,50 +38,59 @@ class NetworkConfig;
 
 namespace client {
 /**
- * @brief This class is responsible for executing the REST requests.
+ * @brief Executes the REST requests.
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]:
```suggestion
 * @brief Executes HTTP request.
```
Not related to the change, as the previous version of the docs was not entirely correct. Would that read a bit simpler and cleaner if it were reading that it executes [any] HTTP requests?

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h [hunk 2]
```diff
@@ -38,50 +38,59 @@ class NetworkConfig;
...
-   * @param query_params Params that is appended to the path url.
-   * @param header_params Headers to customize request.
-   * @param form_params For a POST request, form_params or post_body should be
-   * populated, but not both.
-   * @param post_body For a POST request, form_params or post_body should be
-   * populated, but not both.
-   * This data must not be modified until the request is completed.
-   * @param content_type Content type for the post_body or form_params.
-   * @param callback a function callback to receive the HttpResponse.
+   * @brief Executes the REST request through the network stack.
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]:
```suggestion
   * @brief Executes the HTTP request through the network stack.
```
Not related to the change, as the previous version of the docs was not entirely correct.

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h [hunk 3]
```diff
@@ -32,21 +32,22 @@ class OlpClient;
 }
 
 /**
- * @brief client namespace
+ * @brief The `client` namespace.
  */
 namespace client {
 /**
- * @brief Factory that creates OlpClient instances that is used for all REST
+ * @brief Creates the `OlpClient` instances that are used for all the REST
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h]:
```suggestion
 * @brief Creates the `OlpClient` instances that used for every HTTP request.
```
Not related to the change, as the previous version of the docs was not entirely correct

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h [hunk 4]
```diff
@@ -32,21 +32,22 @@ class OlpClient;
...
  * requests.
  */
 
 class CORE_API OlpClientFactory {
  public:
   /**
-   * @brief Creates an OlpClient instance used for all REST requests.
-   * @param settings Settings for the OlpClient.
+   * @brief Creates the `OlpClient` instance used for all the REST requests.
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h]:
```suggestion
   * @brief Creates the `OlpClient` instance used for every HTTP request.
```
Not related to the change, as the previous version of the docs was not entirely correct

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h [hunk 5]
```diff
@@ -95,17 +104,18 @@ class CORE_API OlpClient {
   /**
    * @brief Executes the REST request through the network stack in a blocking
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]:
```suggestion
   * @brief Executes the HTTP request through the network stack in a blocking
```
Not related to the change but it'd be nice to fix

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 6]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
-   * already cancelled.
-   * @return true in case of successfull execution, false in case context was
+   * @brief Executes the given cancellable code block if the operation is not
    * cancelled.
-   * @deprecated Will be removed once TaskScheduler will be used.
+   *
+   * Otherwise, executes the custom cancellation function.
+   *
+   * @param execute_fn The function that should be executed if this operation is
+   * not cancelled. This function should return `CancellationToken` for which
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: "cancelled" is BrE and "canceled" is AmE.
I know we use AmE for documentation. In this case, I used the BrE spelling to be consistent with the methods and functions that also use the BrE spelling. For example, `IsCancelled`.

So should I change the spelling to AmE or leave it as is (BrE)?

[the remark this answers is not in the document]
Comment (r0busta) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Actually, would it be better if you ignore my suggestions about `cancel` in this PR and make the corrections to the entire SDK in a follow-up PR, so it's consistent and more straightforward to review?

File: olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h [hunk 7]
```diff
@@ -57,34 +67,41 @@ class ApiResponse {
   ApiResponse(const ErrorType& error) : error_(error), success_(false) {}
 
   /**
-   * @brief ApiResponse Copy constructor.
+   * @brief Creates the `ApiResponse` instance that is a copy of the `r`
+   * parameter.
+   *
+   * @param r The `ApiResponse` instance from which the error type is copied.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 8]
```diff
@@ -30,60 +30,83 @@ namespace olp {
 namespace client {
 
 /**
- * @brief Wrapper which manages cancellation state for any asynchronous
- * operations in a thread safe way.
- * All public APIs are thread safe.
+ * @brief A wrapper that manages the cancellation state of an asynchronous
+ * operation in a thread-safe way.
+ *
+ * All public APIs are thread-safe.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Agreed.

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 9]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
    * cancelled.
-   * @deprecated Will be removed once TaskScheduler will be used.
+   *
+   * Otherwise, executes the custom cancellation function.
+   *
+   * @param execute_fn The function that should be executed if this operation is
+   * not cancelled. This function should return `CancellationToken` for which
+   * `CancellationContext` propagates a cancel request.
+   * @param cancel_fn The function that is called if this operation is
+   * cancelled.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Will be changed in a follow-up PR as agreed above.

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 10]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
+   *
+   * @return True if the execution is successful; false if the context was
+   * cancelled.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Will be changed in a follow-up PR as agreed above.

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 11]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
   /**
-   * @brief Allows the user to cancel an ongoing operation in a threadsafe way.
+   * @brief Cancels the ongoing operation in a thread-safe way.
    */
   void CancelOperation();
 
   /**
-   * @brief Check if this context was cancelled.
-   * @return `true` on context cancelled, `false` otherwise.
+   * @brief Checks whether this context is cancelled.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Will be changed in a follow-up PR as agreed above.

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 12]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
+   *
+   * @return True if the context is cancelled; false otherwise.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Will be changed in a follow-up PR as agreed above.

File: olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h [hunk 13]
```diff
@@ -57,34 +67,41 @@ class ApiResponse {
...
+   * @return True if the request is successfully completed; false otherwise.
    */
   inline bool IsSuccessful() const { return success_; }
 
   /**
-   * @brief GetResult Gets the result for a succcessfully executed request
-   * @return A valid Result if IsSuccessful() returns true. Undefined,
+   * @brief Gets the result of the successfully executed request.
+   *
+   * @return A valid result if `IsSuccessful()` returns true; undefined
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h]: Fixed

File: olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h [hunk 14]
```diff
@@ -57,34 +67,41 @@ class ApiResponse {
...
    * otherwise.
    */
   inline const ResultType& GetResult() const { return result_; }
 
   /**
-   * @brief MoveResult Moves the result for a succcessfully executed request
-   * @return A valid Result if IsSuccessful() returns true. Undefined,
+   * @brief Moves the result of the successfully executed request.
+   *
+   * @return A valid result if `IsSuccessful()` returns true; undefined
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h [hunk 15]
```diff
@@ -57,34 +67,41 @@ class ApiResponse {
...
    * otherwise.
    */
   inline ResultType&& MoveResult() { return std::move(result_); }
 
   /**
-   * @brief GetError Gets the error for an unsucccessful request attempt.
-   * @return A valid Error if IsSuccessful() returns false. Undefined,
+   * @brief Gets the error of the unsuccessful request attempt.
+   *
+   * @return A valid error if `IsSuccessful()` returns false; undefined
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h [hunk 16]
```diff
@@ -96,15 +113,17 @@ class ApiResponse {
...
  * it to finalize.
  *
  * @tparam T The result type.
  */
 template <typename T>
 class CancellableFuture {
  public:
-  /// The typedef for the sharable promise.
+  /**
+   * @brief The typedef for the sharable promise.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h [hunk 17]
```diff
@@ -30,60 +30,83 @@ namespace olp {
...
  private:
   /**
-   * @brief Implementation to be able to shared a instance of
-   * CancellationContext.
+   * @brief An implementation used to shared the `CancellationContext` instance.
    */
   struct CancellationContextImpl {
-    /// Mutex lock to protect from concurrent read/write.
+    /**
+     * @brief The mutex lock used to protect from the concurrent read and write
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 18]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
-    Schema,   ///< This HRN represents a schema type.
-    Pipeline  ///< This HRN represents a pipeline instance.
+    Data,     ///< This HRN represents the data catalog.
+    Schema,   ///< This HRN represents the schema type.
+    Pipeline  ///< This HRN represents the pipeline instance.
   };
 
   /**
-   * @brief Constructs a `HRN` from its string form.
+   * @brief Creates the `HRN` instance from its string form.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 19]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    *
-   * The passed string must start with "hrn:".
+   * The passed string must start with `hrn:`.
    *
-   * @param input The `HRN` in string form.
+   * @param input The string representation of this HRN.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 20]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    */
   explicit HRN(const std::string& input);
 
   /**
-   * @brief Constructs a new instance of `HRN` from its string form.
+   * @brief Creates the `HRN` instance from its string form.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 21]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    *
-   * The passed string must start with "hrn:".
+   * The passed string must start with `hrn:`.
    *
-   * @param input The `HRN` in string form.
+   * @param input The string representation of this HRN.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 22]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    *
-   * @return A new instance of `HRN` created from string.
+   * @return The `HRN` instance created from the string.
    */
   static HRN FromString(const std::string& input);
 
   /**
-   * @brief Constructs a unique instance of `HRN` from its string form.
+   * @brief Creates the unique `HRN` instance from its string form.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 23]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    *
-   * The passed string must start with "hrn:".
+   * The passed string must start with `hrn:`.
    *
-   * @param input The `HRN` in string form.
+   * @param input The string representation of this HRN.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 24]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
-   * type relevant fields empty, false otherwise.
+   * @return True if at least one of the service type fields of the `HRN`
+   * instance is empty; false otherwise.
    */
   bool IsNull() const;
 
   /**
-   * @brief Returns this `HRN` instance in its string form, prefixed with
-   * "hrn:".
+   * @brief Converts this HRN to its string representation.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 25]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
+   *
+   * Example: `hrn:partition:service:region:account:resource`
+   *
+   * @return The string representation of this HRN. Prefixed with `hrn:`.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 26]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    */
   std::string ToString() const;
 
   /**
-   * @brief Returns this `HRN` instance in its string form for catalog ID,
-   * prefixed with "hrn:".
+   * @brief Converts this HRN to its string representation for the catalog ID.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HRN.h [hunk 27]
```diff
@@ -28,128 +28,186 @@ namespace olp {
...
    *
-   * @note Only relevant for when the `HRN` has ServiceType == Data.
+   * @note Only relevant if the HRN has `ServiceType == Data`.
+   *
+   * @return The string representation of this HRN for the catalog ID. Prefixed
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HRN.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 28]
```diff
@@ -28,22 +28,40 @@ namespace olp {
 namespace client {
 
 /**
- * @brief Response from a REST request.
+ * @brief A response from a REST request.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 29]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
  */
 class CORE_API HttpResponse {
  public:
   HttpResponse() = default;
+  /**
+   * @brief Creates the `HttpResponse` instance.
+   *
+   * @param status The status of the HTTP response represented.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 30]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
+   * @param response The response to the HTTP request represented as a string.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 31]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
+   */
   HttpResponse(int status, std::string response = std::string())
       : status(status), response(std::move(response)) {}
+  /**
+   * @brief Creates the `HttpResponse` instance.
+   *
+   * @param status The status of the HTTP response.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 32]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
+   * @param response The response to the HTTP request represented as a stream.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 33]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
+   */
   HttpResponse(int status, std::stringstream&& response)
       : status(status), response(std::move(response)) {}
 
   /**
-   * @brief status Status of the HTTP request.
-   * @see olp::network::Network::ErrorCode for negative status numbers
-   * @see olp::network::HttpStatusCode for positive status numbers
+   * @brief The status of the HTTP request.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h [hunk 34]
```diff
@@ -28,22 +28,40 @@ namespace olp {
...
+   *
+   * @see `ErrorCode` for information on negative status
+   * numbers.
+   * @see `HttpStatusCode` for information on positive status
+   * numbers.
    */
   int status{static_cast<int>(olp::http::ErrorCode::UNKNOWN_ERROR)};
+  /**
+   * @brief The response to the HTTP request.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h]: Agree
Changed

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h (hunk 1 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h (hunk 2 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/OlpClient.h (hunk 5 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/OlpClient.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h (hunk 4 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h (hunk 3 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/client/OlpClientFactory.h]: Changed

**Pull request merged**